File: datastructure/06/GradeCounter.java
public class GradeCounter {
    private int _numberOfA;
    private int _numberOfB;
    private int _numberOfC;
    private int _numberOfD;
    private int _numberOfF;


    public int numberOfA() {
        return _numberOfA;
    }

    private void setNumberOfA(int newNumberOfA) {
        this._numberOfA = newNumberOfA;
    }

    public int numberOfB() {
        return _numberOfB;
    }

    private void setNumberOfB(int newNumberOfB) {
        this._numberOfB = newNumberOfB;
    }

    public int numberOfC() {
        return _numberOfC;
    }

    private void setNumberOfC(int newNumberOfC) {
        this._numberOfC = newNumberOfC;
    }

    public int numberOfD() {
        return _numberOfD;
    }

    private void setNumberOfD(int newNumberOfD) {
        this._numberOfD = newNumberOfD;
    }

    public int numberOfF() {
        return _numberOfF;
    }

    private void setNumberOfF(int newNumberOfF) {
        this._numberOfF = newNumberOfF;
    }

    public GradeCounter() { //grade counter setting
        this.setNumberOfA(0);
        this.setNumberOfB(0);
        this.setNumberOfC(0);
        this.setNumberOfD(0);
        this.setNumberOfF(0);
    }

    public void count(char aGrade) {
        switch (aGrade) {
            case 'A':
                this.setNumberOfA(this.numberOfA() + 1);
                break;
            case 'B':
                this.setNumberOfB(this.numberOfB() + 1);
                break;
            case 'C':
                this.setNumberOfC(this.numberOfC() + 1);
                break;
            case 'D':
                this.setNumberOfD(this.numberOfD() + 1);
                break;
            case 'F':
                this.setNumberOfF(this.numberOfF() + 1);
                break;
        }
    }
}

File: practice_cpp/06/GradeCounter.c
#include "GradeCounter.h"
struct _GradeCounter { 
	int numberOfA;
	int numberOfB;
	int numberOfC;
	int numberOfD;
	int numberOfF;
};

GradeCounter* GradeCounter_new() { //생성
	GradeCounter* _this=NewObject(GradeCounter);

	_this->numberOfA=0;
	_this->numberOfB=0;
	_this->numberOfC=0;
	_this->numberOfD=0;
	_this->numberOfF=0;

	return _this;
}

void GradeCounter_delete(GradeCounter* _this) { //소멸
	free(_this);
}

void GradeCounter_count(GradeCounter* _this, char aGrade) {
	switch (aGrade) {
	case 'A':
		_this->numberOfA++;
		break;
	case 'B':
		_this->numberOfB++;
		break;
	case 'C':
		_this->numberOfC++;
		break;
	case 'D':
		_this->numberOfD++;
		break;
	case 'F':
		_this->numberOfF++;
		break;
	}
}

int GradeCounter_numberOfA(GradeCounter* _this) { // a 접근
	return _this->numberOfA;
}

int GradeCounter_numberOfB(GradeCounter* _this) { //b 접근
	return _this->numberOfB;
}
int GradeCounter_numberOfC(GradeCounter* _this) {//c 접근
	return _this->numberOfC;
}
int GradeCounter_numberOfD(GradeCounter* _this) {//d 접근
	return _this->numberOfD;
}
int GradeCounter_numberOfF(GradeCounter* _this) {//e 접근
	return _this->numberOfF;
}